### DISCUSSION TITLE
Register Azure Purview Provider

### DISCUSSION THREAD

**Discussion by zeinab-mk** [Azure/data-management-zone#111] [Q&A] [answered]
It would be nice if registering required Resource Providers for deploying Azure Purview can be included in the deployment templates:

- Microsoft.Purview
- Microsoft.Storage
- Microsoft.EventHub

note: in case you consider including this feature SPN may require additional role.

**Comment by marvinbuss** [comment]
For now we are asking customers to register the Resource Providers themselves. I will  add this to our backlog! Thanks!

**Comment by marvinbuss** [comment]
I will connect to the ESLZ team to discuss this. We will also add this to the new docs as prerequisite.

**Reply by arvind-dhariwal-zz** [reply]
I was about to write it :) Thanks

**Comment by marvinbuss** [comment, marked as answer]
We have added a Notes section here:
https://github.com/Azure/data-management-zone/blob/main/docs/EnterpriseScaleAnalytics-Prerequisites.md#what-will-be-deployed

@zeinab-mk Do you believe that users require more guidance?

**Reply by arvind-dhariwal-zz** [reply]
@marvinbuss Note is good as of now but sometimes customers forget to read that detail and adding RP as part of our internal deployment process,is a good idea in the future.
Also not sure if customers needs their internal approval before adding a new RP.

**Reply by marvinbuss** [reply]
Because of the composition of our templates, RPs get automatically registered, if the user deploys Data Management Zone and Data Landing Zone through the separate "Deploy to Azure"  buttons.
This is not the case for the Enterprise-Scale Analytics "Deploy to Azure" Button. I will look into options on how we can improve this.

@arvind-dhariwal Do you want me to mention all the RPs or just a few in the docs or what are you preferring? Should we also mention it in the "Deploy to Azure" frontend?

**Reply by arvind-dhariwal-zz** [reply]
If we are already listing down or showing the list of services , will be created as part of "Deploy to Azure" then I don't think we need to mention it explicitly. I believe you have already mentioned - https://github.com/Azure/data-management-zone/blob/main/docs/EnterpriseScaleAnalytics-Prerequisites.md#what-will-be-deployed . Answering the question of "What will be deployed" as part of this "Deploy to Azure" is really important to adopt it easily.

**Reply by marvinbuss** [reply]
Ok. Then I will keep it as it is today. Please let me know in this thread if anything needs to be changed. 👍
